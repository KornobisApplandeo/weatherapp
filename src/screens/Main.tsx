import React, {FC, useCallback, useState} from 'react';
import {Image, ScrollView, StyleSheet, Text, View} from 'react-native';
import {IWeatherResponse, searchForWeather, WeatherData} from '../api';
import {BaseTextInput, Button} from '../components';
import WeatherItem from '../components/WeatherItem';
import {colors, FontFamily} from '../types/enums';

const Main: FC = () => {
  const [searchInput, setSerachInput] = useState<string>();
  const [weatherResult, setWeatherResult] = useState<WeatherData[]>();
  const [isLoading, setIsLoading] = useState(false);

  const searchWeather = useCallback(async () => {
    if (searchInput) {
      try {
        setIsLoading(true);
        const response = await searchForWeather(searchInput);
        const jsonResponse: IWeatherResponse = await response?.json();
        if (jsonResponse) {
          setWeatherResult(jsonResponse.list);
        }
        setIsLoading(false);
      } catch (e) {
        console.log('error:', e);
        setIsLoading(false);
      }
    }
  }, [searchInput]);

  return (
    <View style={styles.container}>
      <ScrollView
        contentContainerStyle={styles.scrollview}
        contentInsetAdjustmentBehavior="automatic">
        <View style={styles.bodyWrapper}>
          <View style={styles.titleWrapper}>
            <Text style={styles.title}>Hello Sunshine!</Text>
            <Text style={styles.subtitle}>
              Can you please tell me the weather in Germany?
            </Text>
            <Text style={styles.body}>Please enter a city</Text>
            <BaseTextInput onChangeText={setSerachInput} style={styles.input} />
            <Button
              title="Have a look"
              isLoading={isLoading}
              style={styles.button}
              onPress={searchWeather}
            />
            {weatherResult?.map((item, index) => (
              <View key={index} style={styles.weatherItem}>
                <WeatherItem data={item} />
              </View>
            ))}
          </View>
          <View style={styles.image}>
            <Image source={require('../assets/images/taxi-design.png')} />
          </View>
        </View>
      </ScrollView>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  bodyWrapper: {
    flexDirection: 'column',
    justifyContent: 'space-between',
    flex: 1,
  },
  scrollview: {
    flexGrow: 1,
    justifyContent: 'space-between',
  },
  titleWrapper: {
    paddingHorizontal: 30,
    alignItems: 'center',
    width: '100%',
  },
  title: {
    paddingVertical: 30,
    fontFamily: FontFamily.NUNITO_BLACK,
    fontSize: 30,
    color: colors.orange,
  },
  subtitle: {
    fontFamily: FontFamily.NUNITO_BLACK,
    fontSize: 30,
    textAlign: 'center',
  },
  body: {
    fontFamily: FontFamily.ROBOTO_REGULAR,
    fontSize: 16,
    paddingTop: 30,
    paddingBottom: 15,
  },
  weatherItem: {
    width: '100%',
    paddingBottom: 20,
  },
  input: {
    width: '80%',
    height: 50,
    borderWidth: 1,
  },
  button: {
    marginVertical: 20,
  },
  image: {
    width: 300,
    height: 300,
    overflow: 'hidden',
    alignSelf: 'flex-end',
  },
});

export default Main;
