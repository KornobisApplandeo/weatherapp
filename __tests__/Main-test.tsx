import React from 'react';
import {render} from '@testing-library/react-native';
import Main from '../src/screens/Main';

const setup = () => {
  return render(<Main />);
};

describe('Main screen', () => {
  it('renders without crashing', () => {
    const component = setup().toJSON();
    expect(component).toMatchSnapshot();
  });
});
