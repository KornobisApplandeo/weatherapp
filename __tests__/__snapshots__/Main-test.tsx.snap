// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main screen renders without crashing 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "flexGrow": 1,
        "justifyContent": "space-between",
      }
    }
    contentInsetAdjustmentBehavior="automatic"
  >
    <View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "space-between",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "paddingHorizontal": 30,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#F3997B",
                "fontFamily": "Nunito-Black",
                "fontSize": 30,
                "paddingVertical": 30,
              }
            }
          >
            Hello Sunshine!
          </Text>
          <Text
            style={
              Object {
                "fontFamily": "Nunito-Black",
                "fontSize": 30,
                "textAlign": "center",
              }
            }
          >
            Can you please tell me the weather in Germany?
          </Text>
          <Text
            style={
              Object {
                "fontFamily": "Roboto-Regular",
                "fontSize": 16,
                "paddingBottom": 15,
                "paddingTop": 30,
              }
            }
          >
            Please enter a city
          </Text>
          <TextInput
            onChangeText={[Function]}
            style={
              Object {
                "borderWidth": 1,
                "height": 50,
                "width": "80%",
              }
            }
            textAlign="center"
          />
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#F3997B",
                "borderRadius": 5,
                "marginVertical": 20,
                "opacity": 1,
                "padding": 15,
                "shadowOffset": Object {
                  "height": 20,
                  "width": 20,
                },
                "width": 200,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontFamily": "Roboto-Bold",
                  "fontSize": 12,
                }
              }
            >
              Have a look
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "alignSelf": "flex-end",
              "height": 300,
              "overflow": "hidden",
              "width": 300,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/images/taxi-design.png",
              }
            }
          />
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
